tests/test_connection.py

```diff
@@ -1183,6 +1200,9 @@ def test_send_invalid_tree_id(self, smb_real):
             connection.disconnect()
 
     def test_connection_echo(self, smb_real):
+        """
```

Comment:
I have only added a docstring while trying to understand the scope/purpose of this test.

If you are happy with the new fixture, I can create a separate PR and update all the existing tests to use the fixture and remove the duplicate code.

Reply:
That sounds nice, don't feel obliged to do this though.